test/components/legendTests.ts

```diff
@@ -254,6 +254,43 @@ describe("Legend", () => {
     svg.remove();
   });
 
+  it("can set formatter and change how entry labels are displayed", () => {
+    color.domain(["A", "B", "C"]);
+    let formatter = (id: string) => String.fromCharCode(id.charCodeAt(0) + 1);
+    legend.formatter(formatter);
+    legend.renderTo(svg);
+    let legendRows = svg.selectAll(entrySelector);
+    assert.operator(legendRows.size(), ">=", 1, "There is at least one entry in the test");
+    legendRows.each(function(d, i){
+      let expectText = formatter(legend.colorScale().domain()[i]);
+      assert.strictEqual(d3.select(this).select("text").text(), expectText,
+        `formatter output ${expectText} should be displayed`);
+    });
+    svg.remove();
+  });
+
+  it("can get formatter of the legend using formatter()", () => {
+    let formatter = (id: string) => String.fromCharCode(id.charCodeAt(0) + 1);
```

Comment:
I personally like something even simpler like `(id: string) => \`${id}foo\``

The problem that I have with this formatter is that there are 3 things happening in order to achieve the formatted value, which is just needless overhead if people needed to decipher what happens if a test were to fail with this formatter